code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -471,22 +505,37 @@
 		reagent_id = select_medication(C, beaker_injection)
 
 	if(!reagent_id) //If they don't need any of that they're probably cured!
-		var/list/messagevoice = list("All patched up!" = 'sound/voice/mpatchedup.ogg', "An apple a day keeps me away." = 'sound/voice/mapple.ogg', "Feel better soon!" = 'sound/voice/mfeelbetter.ogg')
-		var/message = pick(messagevoice)
-		speak(message)
-		playsound(loc, messagevoice[message], 50, FALSE)
+		medbot_phrase(pick(finish_healing_phrases), C)
 		bot_reset()
 		return
-	else
-		if(!emagged && !hijacked && check_overdose(patient, reagent_id, injection_amount))
-			soft_reset()
-			return
-		C.visible_message("<span class='danger'>[src] is trying to inject [patient]!</span>", \
-			"<span class='userdanger'>[src] is trying to inject you!</span>")
 
-		addtimer(CALLBACK(src, PROC_REF(do_inject), C, !isnull(beaker_injection), reagent_id), 3 SECONDS)
+	if(!emagged && !hijacked && check_overdose(patient, reagent_id, injection_amount))
+		soft_reset()
 		return
 
+	C.visible_message(
+		"<span class='danger'>[src] is trying to inject [patient]!</span>",
+		"<span class='userdanger'>[src] is trying to inject you!</span>"
+	)
+
+	if(!do_after(src, 3 SECONDS, target = C) || !on || get_dist(src, patient <= 1) || !assess_patient(patient))
```

Comment:
Why was this changed from a callback to a do_after? Explaining why these changes are beneficial in the PR description saves the trouble of having to circle back and respond to such questions in the review.

It also looks like this code is a duplicate of do_inject; if that proc is no longer needed, please remove it

Reply:
The timer callback means theres a strict 3-second lockout to when a bot can begin to act again. If the target moves away quickly, that's 1 SSmobs tick that the bot cannot attempt to either move closer to the target or try to find another one.